includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -718,8 +718,30 @@ public function maybe_process_upe_redirect() {
 				if ( isset( $_GET['redirect_status'] ) && 'succeeded' === $_GET['redirect_status'] ) {
 					$customer = new WC_Stripe_Customer( wp_get_current_user()->ID );
 					$customer->clear_cache();
-
 					wc_add_notice( __( 'Payment method successfully added.', 'woocommerce-gateway-stripe' ) );
+
+					// The newly created payment method does not inherit the customers' billing info, so we manually
+					// trigger an update; in case of failure we log the error and continue because the payment method's
+					// billing info will be updated when the customer makes a purchase anyway.
+					try {
+						$setup_intent_id = isset( $_GET['setup_intent'] ) ? wc_clean( wp_unslash( $_GET['setup_intent'] ) ) : '';
+						$setup_intent = WC_Stripe_API::request( [], 'setup_intents/' . $setup_intent_id, 'GET' );
+
+						$customer_data = WC_Stripe_Customer::map_customer_data( null, new WC_Customer( wp_get_current_user()->ID ) );
+						WC_Stripe_API::request(
+							[
+								'billing_details' => [
+									'name'    => $customer_data['name'],
+									'email'   => $customer_data['email'],
+									'phone'   => $customer_data['phone'],
+									'address' => $customer_data['address'],
+								],
+							],
+							'payment_methods/' . $setup_intent->payment_method
+						);
+					} catch ( Exception $e ) {
+						WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
```

Comment:
```suggestion
					} catch ( WC_Stripe_Exception $e ) {
						WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
						wc_add_notice( $e->getLocalizedMessage(), 'error', [ 'icon' => 'error' ] );
```
This is a minor one, but looks like `WC_Stripe_Customer` and `WC_Stripe_API` both only throw `WC_Stripe_Exception`: we could take advantage of this to throw a nice localised error notice here (or we could not, up to you).

Reply:
I agree with the change to `WC_Stripe_Exception`, but not with adding an error notice... the payment method was added and it will be shown after the page reloads, I think that displaying an error notice to the customer would be confusing.

Also, the missing Stripe info will be filled/updated when the customer makes a payment.

Reply:
Cool, that makes sense. I was thinking of mirroring the notice that is displayed in the error `else` block, but agree catching any exception thrown in the opposite block is already in a successful state and consequently redundant. Considering this further, I think the only exceptions thrown are likely to be due to network errors during API requests and in those scenarios we will probably be unable to add the payment method in the first place, so won't even get that far.